### PR TITLE
Adds a PodDisruptionBudget and topologySpreadConstraints to web

### PR DESCRIPTION
**Pull request by sringel**
## What was changed

- added configurable podDisruptionBudget to web component
- added configurable topologySpreadConstraints to web component
- bugfix: topologySpreadConstraints should be of type list instead of map

## Why?

To enable the web component to have a PodDisruptionBudget created and topologySpreadContraints configured due to the requirement of these configurations being enforced by policy.

## Checklist

1. Closes 

2. How was this tested:

`helm template .`

3. Any docs updates needed?

### CODE REVIEW

File: templates/web-pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{{- if $.Values.web.enabled }}
+{{- if and (gt ($.Values.web.replicaCount | int) 1) ($.Values.web.podDisruptionBudget) }}
```
Comment (robholland) [inline templates/web-pdb.yaml:2]: I would not filter based on replica count. There is a good chance someone might scale outside of helm later.

**Review by robholland — CHANGES_REQUESTED** [review]

File: templates/web-pdb.yaml (hunk 1 above)
Comment (sringel) [reply templates/web-pdb.yaml:2]: This matches the other pdb template
https://github.com/temporalio/helm-charts/blob/master/charts/temporal/templates/server-pdb.yaml#L4

Comment (robholland) [reply templates/web-pdb.yaml]: I'll fix that one in a later PR.

Comment (sringel) [reply templates/web-pdb.yaml]: removed replicaCount check

**Review by robholland — APPROVED** [review]

**Pull request merged**